pkg/manager-nvme/manager.go

```diff
@@ -404,6 +435,35 @@ func (s *Storage) purge() error {
 	return nil
 }
 
+func (s *Storage) purgeExcept(namespaceIdsToKeep []nvme.NamespaceIdentifier) error {
```

Comment:
Be careful here: The call to `purge()` occurs before the storage device is fully initialized. Your call to `purgeExcept`  occurs after init, particularly after the call to `recoverVolumes` has rebuilt the volume list.

So you should be calling `s.deleteVolume()` so the internal Volumes[] are also deleted. Note that takes in a `volumeId`, not a namespace id, so you'll need to redo the API

Reply:
Will we have volumes to delete on storage devices that aren't accounted for by Storage Pools?

Reply:
Yes, `recoverVolumes()` loads all namespaces present on the drives

Reply:
I've changed this to use `ProvidingVolumes`, which contains the `Storage` object and a `VolumeId`. This struct was already present in `storage_pools.go`. I ended up moving it to the NVMe Manager and then changing `purgeExcept()` to `purgeVolumes()`, which now performs the delete with `s.deleteVolume()`.